src/compiler/scanner.ts

```diff
@@ -2251,6 +2254,14 @@ namespace ts {
             return token;
         }
 
+        function reScanPrivateIdentifier(): SyntaxKind {

```

Comment:
I know the various `reScan` functions don’t all have perfectly comparable behavior, but it seems like the general naming convention is naming it for what you _want_, not what you _currently have_: “I currently have token `X`, but have reason to believe that could be interpreted as token `Y`, so to re-scan `X` as `Y`, I call `reScanY`.”

The naming convention is clearly bad when read like that, but diverging from it is probably worse. I would either conform, naming this `reScanHashToken`, or be totally explicit, naming it `reScanPrivateIdentifierAsHashToken`.

Reply:
Nope, I just confused myself. I will reverse it.